### PR TITLE
feat: add batchSaveReplayResult interface

### PR DESCRIPTION
**Pull request by pangdayuan1**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -222,6 +222,12 @@
         <artifactId>redisson</artifactId>
         <version>${reddison.version}</version>
       </dependency>
+
+      <dependency>
+        <groupId>org.apache.maven</groupId>
+        <artifactId>maven-artifact</artifactId>
+        <version>${maven.artifact.version}</version>
```
Comment (QizhengMo) [inline pom.xml:229]: please comment the usage of this dependency

File: arex-storage-web-api/src/main/java/com/arextest/storage/web/controller/ScheduleReplayQueryController.java [hunk 2]
```diff
@@ -333,6 +356,35 @@ public Response cacheRemove(@RequestBody QueryMockCacheRequestType requestType)
     }
   }
 
+  /**
+   * Determine whether to adopt new logic based on the agent version number
+   * @param replayId
+   * @return
+   */
+  private boolean getFromCompareRelation(String replayId) {
```
Comment (QizhengMo) [inline arex-storage-web-api/src/main/java/com/arextest/storage/web/controller/ScheduleReplayQueryController.java:364]: only need to check if the new List key exist or not, not extra key is needed

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/handler/mocker/AbstractAgentWorkingService.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import com.arextest.storage.enums.InvalidReasonEnum;
+import com.arextest.storage.metric.AgentWorkingMetricService;
+import com.arextest.storage.model.InvalidIncompleteRecordRequest;
+import com.arextest.storage.service.InvalidRecordService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public abstract class AbstractAgentWorkingService {
```
[the remark this answers is not in the document]
Comment (pangdayuan1) [reply arex-storage-web-api/src/main/java/com/arextest/storage/service/handler/mocker/AbstractAgentWorkingService.java:13]: Extract some common logic, such as invalidating data when it fails

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/handler/mocker/AgentWorkingHandler.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.arextest.storage.service.handler.mocker;
+
+import java.util.Collections;
+import java.util.List;
+
+public interface AgentWorkingHandler<T> {
```
[the remark this answers is not in the document]
Comment (pangdayuan1) [reply arex-storage-web-api/src/main/java/com/arextest/storage/service/handler/mocker/AgentWorkingHandler.java:6]: Yes, the behavior of saving mocker data will also be extracted later, and if you put it together for review, the amount of code will be a lot, which is not convenient to read

**Review by QizhengMo — APPROVED** [review]

**Review by coryhh — APPROVED** [review]

**Review by QizhengMo — APPROVED** [review]

**Pull request merged**